### ISSUE TITLE
Support VALUES in distributed queries

### ISSUE THREAD

**Issue by sm2017** [citusdata/citus#1837]
I have some VALUES and I want to query it

```sql
SELECT
  max("p"."id") AS "id",
  "p"."postId"
FROM "post" "p"
  LEFT JOIN (VALUES (2, 15119475012016352), (1, 15119471080299999), (3, 15115012036571952)) AS t("postId", "maxId")
  USING ("postId")
WHERE
  "p"."id" < "maxId"
  AND
  "p"."postId" IN (2, 1, 3)
GROUP BY "p"."postId";
```

I change it to the following

```sql
WITH t("postId", "maxId") AS (VALUES (2, 15119475012016352), (1, 15119471080299999), (3, 15115012036571952))
SELECT
  max("p"."id") AS "id",
  "p"."postId"
FROM "post" "p"
  LEFT JOIN t USING ("postId")
WHERE
  "p"."id" < "maxId"
  AND
  "p"."postId" IN (2, 1, 3)
GROUP BY "p"."postId"
```

both have the following error

> [0A000] ERROR: could not run distributed query with complex table expressions
> Hint: Consider using an equality filter on the distributed table's partition column.

I don't want use AND , OR in my query instead of joining as I have many VALUES and many AND  OR is very slower than simple joining

How can I do in in application layer (Without procedure)

**Comment by onderkalaci** [comment]
Citus currently doesn't support `VALUES`. Actually, it is a lot simpler to support after #1805. We might consider adding that in the future.

I can suggest a workaround for now (changed the schema to fit my dev env.):

```SQL
CREATE TYPE myrowtype AS (f1 int, f2 bigint);
SELECT run_command_on_workers('CREATE TYPE myrowtype AS (f1 int, f2 bigint);')  


SELECT
  max("p"."value_2") AS "value_2",
  "p"."user_id"
FROM "users_table" "p"
  LEFT JOIN (SELECT  * FROM unnest(ARRAY[row(2, 15119475012016352)::myrowtype, row(1, 15119471080299999)::myrowtype, row(3, 15115012036571952)::myrowtype])) AS t("user_id", "maxId")
  USING ("user_id")
WHERE
  "p"."value_2" < "maxId"
  AND
  "p"."user_id" IN (2, 1, 3)
GROUP BY "p"."user_id";

```

**Comment by marcocitus** [comment]
Let's not close this issue just yet. We could implement VALUES.

**Comment by marcocitus** [comment]
> Actually, it is a lot simpler to support after #1805. We might consider adding that in the future.

Except we'd have to stop abusing `values_lists` in RTEs :).

The good news is that VALUES will be automatically supported when we're done with #1804.

**Comment by marcocitus** [comment]
> `WITH t("postId", "maxId") AS (VALUES (2, 15119475012016352), (1, 15119471080299999), (3, 15115012036571952))
> ...`

The query is now supported on the master branch

**Comment by marcocitus** [comment]
Fixed by #1876
